examples/scott_dick_controller.py

```diff
@@ -0,0 +1,251 @@
+        
+
+        # My demonstration controller does not move the ship, only rotates it to shoot the nearest asteroid.
+        # Goal: demonstrate processing of game state, fuzzy controller, intercept computation 
+        # Intercept-point calculation derived from the Law of Cosines, see notes for details and citation.
+
+        # Find the closest asteroid (disregards asteroid velocity)
+        ship_pos_x = ship_state["position"][0]     # See src/kesslergame/ship.py in the KesslerGame Github
+        ship_pos_y = ship_state["position"][1]       
+        closest_asteroid = None
+        
+        for a in game_state["asteroids"]:
+            #Loop through all asteroids, find minimum Eudlidean distance
+            curr_dist = math.sqrt((ship_pos_x - a["position"][0])**2 + (ship_pos_y - a["position"][1])**2)
+            if closest_asteroid is None :
+                # Does not yet exist, so initialize first asteroid as the minimum. Ugh, how to do?
+                closest_asteroid = dict(aster = a, dist = curr_dist)
+                
+            else:    
+                # closest_asteroid exists, and is thus initialized. 
+                if closest_asteroid["dist"] > curr_dist:
+                    # New minimum found
+                    closest_asteroid["aster"] = a
+                    closest_asteroid["dist"] = curr_dist
+
+        # closest_asteroid is now the nearest asteroid object. 
+        # Calculate intercept time given ship & asteroid position, asteroid velocity vector, bullet speed (not direction).
+        # Based on Law of Cosines calculation, see notes.
+        
+        # Side D of the triangle is given by closest_asteroid.dist. Need to get the asteroid-ship direction
+        #    and the angle of the asteroid's current movement.
+        # REMEMBER TRIG FUNCTIONS ARE ALL IN RADAINS!!!
+        
+        
+        asteroid_ship_x = ship_pos_x - closest_asteroid["aster"]["position"][0]
+        asteroid_ship_y = ship_pos_y - closest_asteroid["aster"]["position"][1]
+        
+        asteroid_ship_theta = math.atan2(asteroid_ship_y,asteroid_ship_x)
+        
+        asteroid_direction = math.atan2(closest_asteroid["aster"]["velocity"][1], closest_asteroid["aster"]["velocity"][0]) # Velocity is a 2-element array [vx,vy].
+        my_theta2 = asteroid_ship_theta - asteroid_direction
+        cos_my_theta2 = math.cos(my_theta2)
+        # Need the speeds of the asteroid and bullet. speed * time is distance to the intercept point
+        asteroid_vel = math.sqrt(closest_asteroid["aster"]["velocity"][0]**2 + closest_asteroid["aster"]["velocity"][1]**2)
+        bullet_speed = 800 # Hard-coded bullet speed from bullet.py
+        
+        # Determinant of the quadratic formula b^2-4ac
+        targ_det = (-2 * closest_asteroid["dist"] * asteroid_vel * cos_my_theta2)**2 - (4*(asteroid_vel**2 - bullet_speed**2) * closest_asteroid["dist"])

```

Comment:
Mostly looks good!

closest_asteroid["dist"] needs to be squared here. This mathematical error will prevent precise aiming of asteroids, and is the cause of the "aim lag" described in the PDF guide.

Additionally, there should be no need for a PID controller, as we are directly in control of the ship's rotation rate on each frame and can change this instantaneously. A PID controller would only be useful if we only had control of the angular acceleration of the ship.